### ISSUE TITLE
common mistakes in pytorch

### ISSUE THREAD

**Issue by visionNoob** [insurgent92/InsightFace-v2#1]

**Comment by visionNoob** [comment]
most common neural net mistakes: 

1) you didn't try to overfit a single batch first. 
2) you forgot to toggle train/eval mode for the net. 
3) you forgot to .zero_grad() (in pytorch) before .backward(). 
4) you passed softmaxed outputs to a loss that expects raw logits. ; others? :)
5) you didn't use bias=False for your Linear/Conv2d layer when using BatchNorm, or conversely forget to include it for the output layer 
6) thinking view() and permute() are the same thing (& incorrectly using view)

reference 
https://twitter.com/karpathy/status/1013244313327681536
